client/src/components/exercise/lab10/pages/TrainingAI.js

```diff
@@ -0,0 +1,151 @@
+import React, { Fragment, useEffect, useState } from "react";
+import { bindActionCreators } from "redux";
+import { actions as exerciseActions } from "../../../../reducers/lab10/ExerciseReducer";
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
+import Simulation from "../components/Simulation";
+import {
+  EXERCISE_PLAYING,
+  MIN_USER_ATTEMPTS,
+  SIMULATION_ENDED,
+  SIMULATION_IDLE,
+  SIMULATION_STARTED,
+} from "../../../../constants/lab10";
+import useScroll from "../../../../use-hooks/useScroll";
+import { navigate } from "@reach/router";
+
+const TrainingAI = (props) => {
+  useScroll();
+  const [limitReached, setLimitReach] = useState(false);
+
+  /**
+   * Executed on mount
+   */
+  useEffect(() => {
+    props.actions.updateState(EXERCISE_PLAYING);
+    props.actions.disableSimulationCover();
+    props.actions.enableUserInput();
+    props.actions.idleSimulation();
+  }, []);
+
+  /**
+   * Enable/Disable properties based on the SIMULATION state
+   */
+  useEffect(() => {
+    switch (props.simulationStatus) {
+      case SIMULATION_STARTED:
+        props.actions.enableUserInput();
+        props.actions.disableSimulationCover();
+        break;
+      case SIMULATION_ENDED:
+        break;
+    }
+  }, [props.simulationStatus]);
+
+  /**
+   * Update state if the minimum amount user attempts has been reached
+   */
+  useEffect(() => {
+    if (props.userAttempts >= MIN_USER_ATTEMPTS && !limitReached) {
+      setLimitReach(true);
+      props.actions.disableUserInput();
+      props.actions.enableSimulationCover();
+    }
+  }, [props.userAttempts]);
+
+  /**
+   * Redirect the user to the following page
+   */
+  const handleContinue = () => {
+    return navigate("/Lab10/Exercise/TrainingAI/Repair");
+  };
+
+  return (
+    <div>
+      {props.simulationStatus === SIMULATION_IDLE && (
+        <div>
+          {limitReached ? (
+            <Fragment>
+              <p className={"playthrough__sentence"}>
+                Awesome, it works. Next, you need to generate data for the
+                neural network. This data can be generated by training the AI to
+                avoid the falling shapes, which will be the goal for this phase
+                of the exercise. <br /> You will be given{" "}
```

Comment:
`of the exercise. <br /> You will be given{" "}` why do you need the empty quotes section

Reply:
since the value is on the next line, {" "} allows a space between `given` and `props.trainingDuration`. otherwise, they would be together.